lib/elixir/lib/integer.ex

```diff
@@ -400,17 +400,23 @@ defmodule Integer do
       iex> Integer.gcd(10, 0)
       10
 
+      iex> Integer.gcd(7, 7)
+      7
+
       iex> Integer.gcd(0, 0)
       0
 
   """
+  @spec gcd(0, 0) :: 0
+  @spec gcd(int, int) :: int when int: pos_integer
```

Comment:
If we replace `tuple()` by `number()`, than the docs would read:

> The specification with the X type variable specifies that the function takes a number and returns **the same** number.

Wouldn't that mean equality? If so, either the docs are wrong (or extremely confusing) or it does enforce equality.